pitop/common/configuration_file.py

```diff
@@ -0,0 +1,86 @@
+from pathlib import Path
+from re import compile
+
+
+def add_section(
+    filename: str, title: str, description: str, content: str, notes: str = ""
+) -> None:
+    """Add a section to a configuration file with the following format:
+
+    .. code-block:: python
+
+        # [<TITLE>] START
+        #
+        # <DESCRIPTION>
+        #
+        # Notes:
+        # <NOTES>
+        #
+        <CONTENT>
+        #
+        # [<APP_NAME>] END
+    """
+    text = f"""
+# [{title}] START
+#
+# {description}
+#
+# Notes:
+# {notes}
+#
+{content}
+#
+# [{title}] END"""
+    with open(filename, "a") as f:
+        f.write(text)
+
+
+def remove_section(filename: str, title: str) -> None:
+    """Removes a section of the provided title from the given filename. It
+    assumes that the format of the block to remove is the following:
+
+    .. code-block:: python
+
+        # [<TITLE>] START
+        #
+        # <DESCRIPTION>
+        #
+        # Notes:
+        # <NOTES>
+        #
+        <CONTENT>
+        #
+        # [<APP_NAME>] END
+    """
+    start_pattern = compile(rf"^(# \[{title}\] START)\n")
+    finish_pattern = compile(rf"^(# \[{title}\] END)\n")
+    delete_line = False
+
+    with open(filename, "r") as fr:
+        lines = fr.readlines()
+        with open(filename, "w") as fw:
+            for i, line in enumerate(lines):
+                if (
+                    line.strip() == ""
```

Comment:
why should the line before the start pattern have to be empty? when you create the section you don't add an empty line before it?

Reply:
yeah, the string is:
```
text = f"""
# ...
```
which counts as a new line

Reply:
right. it might still be better to drop the requirement for this empty line. what if someone edits the file manually and removes it, it doesn't look important. 

as they say, if `start_pattern.search(lines[i + 1]) is not None`, who cares whether `line.strip() == ""`

Reply:
makes sense - just updated it!

Reply:
tbh i thought you could remove the check for an empty line since it's non-essential, but leave the actual line just for better formatting